components/site/patterns/DeleteTable.js

```diff
@@ -5,59 +5,68 @@ import { Button } from '@twilio-paste/core/button';
 import { DeleteIcon } from '@twilio-paste/icons/cjs/DeleteIcon';
 import PropTypes from 'prop-types';
 
-export const DeleteTable = ({ data, handleDelete }) => {
-  if (data.length === 0) {
+export const DeleteTable = ({ columns, rows, emptyState, handleDelete }) => {
+  if (rows.length === 0) {
     return (
       <Text as="span" fontFamily="fontFamilyText">
-        Nothing to delete!
+        {emptyState}
       </Text>
     );
   }
 
-  const cols = Object.keys(data[0]);
+  rows.forEach((row) => {
+    if (row.length > columns.length)
+      throw new Error(
+        'If you are seeing this error, it is because you added more cells than there are columns. Go back to delete.js and add a new column heading. DO NOT try to fix this error in DeleteTable.js!',
+      );
+  });
 
   return (
     <Table>
       <THead>
         <Tr>
-          {cols.map((key) => {
-            if (key === 'id') return <></>;
-            return <Th key={cols.indexOf(key)}>{key}</Th>;
+          {columns.map((column) => {
+            return <Th key={columns.indexOf(column)}>{column}</Th>;
           })}
           <Th>Actions</Th>
         </Tr>
       </THead>
       <TBody>
-        {data.map((service) => {
-          const row = [];
-          for (let i = 1; i < cols.length; i++) {
-            row.push(<Td key={service.id + i}>{service[cols[i]]}</Td>);
+        {rows.map((row, index) => {
+          if (row.length < columns.length) {
+            const diff = columns.length - row.length;
+            for (let i = 0; i < diff; i++) {
+              row.push(<Text as="span" fontFamily="fontFamilyText" />);
+            }
```

Comment:
To clarify, this is making the row the right size (as long as the columns array) by pushing in empty <Text> components where there was previously nothing, correct?

Reply:
this was to ensure that a "column" of empty cells would be created if a new column was added to the column array. however, simon's suggestion below with mapping over the columns and then mapping over the rows actually takes care of this!